javascript/code-challenges/fifoAnimalShelter/fifo-anmimal-shelter.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+class Animal {
+  constructor(name) {
+    this.name = name;
+    this.next = null;
+    }
+    
+    class AnimalShelter extends Animal {
+      constructor() {
+        this.front = null;
+        this.end = null;
+      }
+      
+      enqueue(animal) {
+        const newAnimal = new Animal(animal);
+        if(!this.front) {
+          this. front = newAnimal;
+          this.end = this.front
+        } else { 
+          this.end.next = newAnimal;
+          this.end = newAnimal;
+        }
+      }
+          
+          dequeue(pref)
+  }
+}
```

Comment:
You're off to a really great start Tahmina! Once you have some time to come back to this finish out code and tests I will regrade for you!